### ISSUE TITLE
DM new project reviews to team members

### ISSUE THREAD

**Issue by jeffreywescott** [LearnersGuild/echo#478]
@jeffreywescott commented on [Tue Aug 09 2016](https://github.com/LearnersGuild/game-prototype/issues/19)
## Benefits

Benefits transparency, feedback to learners.
## Description

When a new _external_ project review (see #501) is submitted, the project team members should each be informed via direct message:
- who the review is from
- what the new external review scores are
- what the impact on aggregate review scores is
## Mechanics Change
- Remove general "review completed" notification to chapter channel (redundant)
- New DM to all members of project when new external review for project is submitted
## UI Design Proposal

DM message format:

```
**New review for #${project.name}**:

Review from ${review.reviewerHandle}
Completeness: ${review.completeness}
Quality: ${review.quality}

**Updated aggregate review scores:**

${project.noOfReviews} reviews submitted.

Avg. completeness: ${project.avgProjCompleteness}
Completeness scores: ${project.completenessReviews.join(' | ')}

Avg. quality: ${project.avgProjQuality}
Quality scores: ${project.qualityReviews.join(' | ')}
```
## Design Notes
- Using a DM to each player of a project instead of a message to the project channel itself, because there is a non-zero chance that project reviews will be submitted at a date beyond the completion of the project, when the project channel may no longer exist.

---

@shereefb commented on [Mon Aug 08 2016](https://github.com/LearnersGuild/learning-os-software/issues/166)

---

@tannerwelsh commented on [Tue Aug 16 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-240156125)

in game mechanics because unanswered is "when does this happen?"

---

@tannerwelsh commented on [Tue Aug 16 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-240157407)

project should know:
- who on their team reviewed and what the scores are
- what the external review scores are (but NOT showing names)
- what the aggregate review scores are

---

@tannerwelsh commented on [Tue Aug 16 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-240158274)

@LearnersGuild/los mind taking a look at the suggested mechanics here? want to move this to engineering board soon

---

@tannerwelsh commented on [Tue Aug 16 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-240161444)

oops, nevermind. moving back for now until #20 is closed

---

@tannerwelsh commented on [Wed Sep 14 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-247105404)

update language and then move to RFI

---

@tannerwelsh commented on [Wed Sep 21 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-248747342)

Moved to review, assigned to @shereefb 

---

@shereefb commented on [Wed Sep 21 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-248781072)

related #20 

---

@shereefb commented on [Wed Sep 21 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-248781106)

lgtm rfi /cc @jeffreywescott 

---

@jeffreywescott commented on [Wed Sep 21 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-248805513)

Process-check: ideally, other stakeholders would have a chance to review this here before it slides into RFI. Are we sure that Software Engineering doesn't need to review this?

---

@shereefb commented on [Thu Sep 22 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-248920192)

significantly altered by #68 

---

@tannerwelsh commented on [Thu Sep 22 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-248959517)

Sorry for the confusion @jeffreywescott - there's some missing pieces here that I need to clarify before it's ready for review. And then, yes, will definitely leave in review for full team feedback.

---

@tannerwelsh commented on [Thu Sep 22 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249014607)

ready for review @shereefb @LearnersGuild/software 

---

@shereefb commented on [Fri Sep 23 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249118205)
- Curious why not reveal the name? That feels a little too protective. Learners should get used to giving/getting feedback on work product with names attached. My vote is reveal the name. 
- This is related to #71 . Does it make sense to hold off on implementing this feature, get #71 in place first, and then have this DM include changes to review-bias stats?
- Projects right now get externally reviewed 6 to 10 times. Is that a lot of spam in the project channel? 
- Oakland channel becomes useless during project reviews because of all the review notifications, can we get rid of them as part of this issue, since it makes them obsolete. 
- As a player, I want to see how the completeness/quality score of this project was impacted by this reviewer, I also want to see how I rated this project compared to how this reviewer rated it. 

Can we list how the team internally reviewed the project, compared how this external reviewer saw it?

Jeffrey said: 80% complete, 95% quality

Your team:
Tanner: 80% C, 95% Q
NeEddra: 94% C, 90% Q

---

@tannerwelsh commented on [Fri Sep 23 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249245642)

> Curious why not reveal the name?

I thought that was your constraint. I didn't write it :) So yes, let's include names. Updated description.

> Does it make sense to hold off on implementing this feature, get #71 in place first, and then have this DM include changes to review-bias stats?

It makes sense from a "keep the game balanced" standpoint, but not from a continuous delivery standpoint. I'd rather get this out there, and then balance it.

> Projects right now get externally reviewed 6 to 10 times. Is that a lot of spam in the project channel?

Whoops, I left out a complete edit of this description. My proposal is to use direct messages to the team members, not to a project channel (see Design Notes in original comment).

> can we get rid of them as part of this issue, since it makes them obsolete.

Yes. Updated issue.

> As a player, I want to see how the completeness/quality score of this project was impacted by this reviewer, I also want to see how I rated this project compared to how this reviewer rated it.

This is not addressed here because I didn't want each message to repeat redundant information. The internal project reviews would be included in the post-retro DM (as long as #68 is accepted), so players could compare the new numbers to the previous message.

---

@shereefb commented on [Fri Sep 23 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249334417)

lgtm.
ship it!

---

@tannerwelsh commented on [Mon Sep 26 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249589712)

Thoughts @prattsj @bundacia? Otherwise gonna move to RFI. :)

---

@bundacia commented on [Mon Sep 26 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249592591)

@tannerwelsh: Sending this kind of info in a DM feels awkward to me. Is there some reason we're not providing a web interface where they can go and look at this stuff? Is this just an assumption that the DM ui is faster to build/iterate?

---

@prattsj commented on [Mon Sep 26 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249644685)

@bundacia @tannerwelsh: +1 for eventually making this into a "pull" UX (e.g. `/stats` or `/reviews` with flex panel) rather than a "push" situation (via DM). DM'd reviews would probably be a bit easier to implement since we're already doing it for stats and individual player feedback, but ideally, longer term, they can just retrieve this data on demand. The biggest challenge in doing it that way that comes to mind is just designing for a long history of project-related information. If we can put something together with a standard drill-down navigation model (project list > project detail), i'd say that's ultimately the more useful UX. but i'm not opposed to starting off with the DM'd approach.

---

@prattsj commented on [Mon Sep 26 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249646950)

tl;dr - LGTM

---

@tannerwelsh commented on [Tue Sep 27 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-249868009)

Thanks for the feedback folks. I'm working on a more comprehensive revamp of the overall UI, but in the meantime let's keep this piece moving forward. DM is not ideal, but better than nothing.

Also note that this is not a high-priority issue (marked as v3 milestone). 

---

@jeffreywescott commented on [Thu Oct 06 2016](https://github.com/LearnersGuild/game-prototype/issues/19#issuecomment-252103247)

Moving to implementation board despite the fact that it's a milestone v3 / not critical for q4. It will be low-priority in the Backlog.
